client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -80,7 +97,29 @@ function getCommentDecorator(commentApp) {
     const blockKey = children[0].props.block.getKey()
     const start = children[0].props.start
 
-    const commentId = useMemo(() => parseInt(contentState.getBlockForKey(blockKey).getInlineStyleAt(start).find((style) => style.startsWith(COMMENT_STYLE_IDENTIFIER)).slice(8)), [blockKey, start]);
+    const commentId = useMemo(() => 
+      {
+        const block = contentState.getBlockForKey(blockKey);
+        const styles = block.getInlineStyleAt(start).filter((style) => style.startsWith(COMMENT_STYLE_IDENTIFIER));
+        let styleToUse;
+        if (styles.count() > 1) {
+          // We're dealing with overlapping comments.
+          // Find the least frequently occurring style and use that - this isn't foolproof, but in most cases should ensure that all comments
+          // have at least one clickable section. This logic is a bit heavier than ideal for a decorator given how often we are forced to
+          // redecorate, but will only be used on overlapping comments
+          const styleFreq = styles.map((style) => {
+            let counter = 0
+            findCommentStyleRanges(block, () => {counter = counter + 1}, (metadata) => metadata.getStyle().some(rangeStyle => rangeStyle === style));
+            return [style, counter]
+          }).sort((a, b) => a[1] - b[1])
+        
+          styleToUse = styleFreq.first()[0];
+
+        } else {
+          styleToUse = styles.first();
+        }
+        return parseInt(styleToUse.slice(8));
```

Comment:
suggestion: I don’t think `8` as a magic value is understandable enough

Reply:
Agreed, I'll change this!